### ISSUE TITLE
Backport proposer based timestamps

### ISSUE THREAD

**Issue by evan-forbes** [celestiaorg/celestia-core#1255]
PBTS have been a long awaited feature. This block validity rule would stop indiviual validators from proposing a block faster than expected, and it would stop a malicious 1/3 from accelerating the recorded block time (time listed in the header, not the actual time between blocks!).

PBTS was originally proposed and implemented years ago, so it might actually be easier to backport that instead of the one recentily added to CometBFT.

It has been speculated that using proposer based timestamps would also fix the issues encountered the last time we tried to make the timeout commit dynamic https://github.com/celestiaorg/celestia-core/pull/965

**Comment by rootulp** [comment]
Can you please include links to the two distinct implementations you're referring to?

**Comment by evan-forbes** [comment]
yeah sure, the first is the one from two years ago https://github.com/cometbft/cometbft/tree/wb/proposal-timestamp-difference

the second will be included in the v1 version of CometBFT, which I couldn't find if it exists. After that version is released, depending on what version our branched is based on we might want to use it instead

**Comment by rootulp** [comment]
Thanks. It's kinda difficult to view the diff on that branch so this PR may also help: https://github.com/tendermint/tendermint/pull/7605
